app/src/main/java/com/picone/go4lunch/presentation/ui/MainActivity.java

```diff
@@ -21,9 +29,34 @@ protected void onCreate(Bundle savedInstanceState) {
         setUpNavigation();
     }
 
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+
+            finish();
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
     public void setUpNavigation() {
         NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
         assert navHostFragment != null;
-        NavigationUI.setupWithNavController(mBinding.bottomNavigation, navHostFragment.getNavController());
+        NavController navController = navHostFragment.getNavController();
+        NavigationUI.setupWithNavController(mBinding.bottomNavigation, navController);
+        NavigationUI.setupWithNavController(mBinding.toolbar, navController);
+        mBinding.toolbar.setTitle(" ");
+
+        navController.addOnDestinationChangedListener((controller, destination, arguments) -> {
+            if (destination.getId() != R.id.authenticationFragment) {
```

Comment:
This block of code can be factorized through a private method.

Reply:
ok done